src/main/java/com/starinfo/StarInfoPlugin.java

```diff
@@ -247,6 +256,24 @@ public void onNpcSpawned(NpcSpawned event)
 		refresh();
 	}
 
+	@Subscribe
+	public void onVarbitChanged(VarbitChanged event)
+	{
+		if (event.getVarbitId() != VARBIT_STAR_DISCOVERY)
+		{
+			return;
+		}
+
+		bonusCount = event.getValue();
+
+		if (!starConfig.showStarDiscovery() || bonusCount == 0)
```

Comment:
Currently, if showStarDiscovery is disabled, this if statement calls removeBonusCounter when bonusCounter is null and probably has a null pointer exception triggered. Check showStarDiscovery first and return with no action so the bonuscount == 0 check is by itself and won't cause this issue.